source/es5-wrappers.ts

```diff
@@ -0,0 +1,124 @@
+import { Block } from "./models/block";
+import { Log } from "./models/log";
+import { BlockHistory } from "./models/block-history";
+import { LogHistory } from "./models/log-history";
+import { BlockAndLogHistory } from "./models/block-and-log-history";
+import { Filter, FilterOptions } from "./models/filters";
+import { reconcileBlockHistory } from "./block-reconciler";
+import { reconcileLogHistoryWithAddedBlock, reconcileLogHistoryWithRemovedBlock } from "./log-reconciler";
+import { reconcileBlocksAndLogs } from "./block-and-log-reconciler";
+
+export function reconcileBlockHistoryEs5(
+	getBlockByHash: (hash: string, callback: (error?: Error, block?: Block | null) => void) => void,
+	blockHistory: BlockHistory | null,
+	newBlock: Block,
+	onBlockAdded: (block: Block, callback: (error?: Error) => void) => void,
+	onBlockRemoved: (block: Block, callback: (error?: Error) => void) => void,
+	blockRetention: number,
+	callback: (error?: Error, newHistory?: BlockHistory) => void,
+): void {
+	const wrappedGetBlockByHash = (hash: string): Promise<Block | null> => new Promise<Block | null>((resolve, reject) => {
+		getBlockByHash(hash, (error, block) => {
+			if (error) reject(error);
+			else resolve(block);
+		});
+	});
+	const wrappedOnBlockAdded = (block: Block): Promise<void> => new Promise<void>((resolve, reject) => {
+		onBlockAdded(block, (error) => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	const wrappedOnBlockRemoved = (block: Block): Promise<void> => new Promise<void>((resolve, reject) => {
+		onBlockRemoved(block, (error) => {
+			if (error) reject(error);
+			else resolve();
+		});
+	});
+	reconcileBlockHistory(wrappedGetBlockByHash, blockHistory, newBlock, wrappedOnBlockAdded, wrappedOnBlockRemoved, blockRetention)
+		.then(newBlockHistory => callback(undefined, newBlockHistory))
+		.catch(error => callback(error, undefined));
+}
+
+export function reconcileLogHistoryWithAddedBlockEs5(
```

Comment:
No, I didn't bother since the contents of that file are just copy/paste from `reconcileBlocksAndLogsWithCallback` and there is no business logic in the function (it is just a thin wrapper around the promise version of the method, which is fully tested.